awx/main/analytics/core.py

```diff
@@ -146,30 +146,26 @@ def gather(dest=None, module=None, collection_type='scheduled'):
 
 def ship(path):
     """
-    Ship gathered metrics via the Insights agent
+    Ship gathered metrics via the Insights API
     """
     try:
-        agent = 'insights-client'
-        if shutil.which(agent) is None:
-            logger.error('could not find {} on PATH'.format(agent))
-            return
         logger.debug('shipping analytics file: {}'.format(path))
-        try:
-            cmd = [
-                agent, '--payload', path, '--content-type', settings.INSIGHTS_AGENT_MIME
-            ]
-            output = smart_str(subprocess.check_output(cmd, timeout=60 * 5))
-            logger.debug(output)
-            # reset the `last_run` when data is shipped
-            run_now = now()
-            state = TowerAnalyticsState.get_solo()
-            state.last_run = run_now
-            state.save()
-
-        except subprocess.CalledProcessError:
-            logger.exception('{} failure:'.format(cmd))
-        except subprocess.TimeoutExpired:
-            logger.exception('{} timeout:'.format(cmd))
+        base_url = getattr(settings, 'AUTOMATION_ANALYTICS_URL', None)
+        if not base_url:
+            logger.execption('AUTOMATION_ANALYTICS_URL is not set')
+        url = base_url + '/api/ingress/v1/upload'
+        with open(path, 'rb') as f:
+            files = {'file': (os.path.basename(path), f, settings.INSIGHTS_AGENT_MIME)}
+            creds = Credential.objects.get(name=settings.INSIGHTS_URL_BASE)
+            response = requests.post(url, files=files, auth=(getattr(settings, 'CLOUD_REDHAT_USERNAME', None),
```

Comment:
Given the content of this data, and for completeness, it would probably be smart to explicitly specify `requests.post(..., verify=True)`.